packages/bruno-electron/src/store/preferences.js

```diff
@@ -189,7 +189,7 @@ const preferencesUtil = {
   markAsLaunched: () => {
     const preferences = getPreferences();
     preferences.onboarding.hasLaunchedBefore = true;
-    preferencesStore.savePreferences(preferences);
+    savePreferences(preferences);
   }
```

Comment:
I would suggest making `markAsLaunched` `async` and `await`ing `savePreferences()`.
This ensures the `hasLaunched` flag is persisted before callers proceed, avoiding race conditions where the app assumes the flag is saved but the write is still in-flight. Also recommend updating call sites to either `await` it or explicitly fire-and-forget with `void`.

```suggestion
markAsLaunched: async () => {
  const preferences = getPreferences();
  preferences.onboarding.hasLaunchedBefore = true;

  try {
    await savePreferences(preferences);
  } catch (err) {
    console.error('Failed to save preferences in markAsLaunched:', err);
  }
}
```